src/ShopifyApp/Middleware/AuthShop.php

```diff
@@ -48,10 +48,17 @@ protected function validateShop(Request $request)
         $session = new ShopSession();
 
         // Grab the shop's myshopify domain from query or session
-        $shopDomainParam = $request->get('shop');
+        // For SPA's we need X-Shop-Domain : See issue https://github.com/ohmybrew/laravel-shopify/issues/295
+        $shopDomainParam = $request->get('shop') ?? $request->header('X-Shop-Domain');
```

Comment:
If you really want to send the domain in the headers you should really also send the HMAC and verify if to prevent spoofing.

So send the both the shop and the signature retrieved from the headers to the `api()->verify request()` call with something like `['shop' => 'example.com', 'hmac' => 'abc123']`